usr/src/uts/common/fs/zfs/sys/spa.h

```diff
@@ -326,7 +327,8 @@ typedef enum bp_embedded_type {
 typedef struct blkptr {
 	dva_t		blk_dva[SPA_DVAS_PER_BP]; /* Data Virtual Addresses */
 	uint64_t	blk_prop;	/* size, compression, type, etc	    */
-	uint64_t	blk_pad[2];	/* Extra space for the future	    */
+	uint32_t	blk_iv[3];	/* IV for block level encryption    */
+	uint32_t	blk_pad;	/* Extra space for the future	    */
```

Comment:
I don't know SMB3 specifically, but I assume what they do there is they completely replace the checksum with the MAC. In ZFS, we use a 256 bit checksum for data, but the maximum length of the MAC (according to the AES-GCM and AES-CCM docs) is 128 bits. The checksum for encrypted datasets is sha256 + mac, which is essentially 128 bits of MAC and 128 bits of regular sha256 of the ciphertext.